### DISCUSSION TITLE
Loosing type annotation between `QueryProcedure` and `DecoratedQuery` (trpc@11.0.0-rc.660)

### DISCUSSION THREAD

**Discussion by michaltaberski** [trpc/trpc#6314] [Q&A] [answered]
I think this 20s video explains the issue well: https://share.cleanshot.com/2JVY3SlZ

I wrote a `TrpcProvider` but it's totally basic, default one:

```jsx
export const trpc = createTRPCReact<AppRouter>()

export const TrpcProvider = ({ children }: { children: React.ReactNode }) => {
  const [queryClient] = useState(() => new QueryClient())
  const [trpcClient] = useState(() =>
    trpc.createClient({
      links: [httpBatchLink({ url: "http://localhost:3000/trpc" })],
    }),
  )

  return (
    <trpc.Provider client={trpcClient} queryClient={queryClient}>
      <QueryClientProvider client={queryClient}>{children}</QueryClientProvider>
    </trpc.Provider>
  )
}

```

```
    "@tanstack/react-query": "^5.62.7",
    "@trpc/react-query": "11.0.0-rc.660",
```
    
I am running it in the context of monorepo, but that shouldn't matter since `AppRouter` type is passed well 🤔 

```
"@trpc/server": "11.0.0-rc.660"
```

Let me know if there is something else I could provide to make the issue easier to debug. Thanks!

**Comment by michaltaberski** [comment, marked as answer]
We can close this ticket, but for those who are interested in fix (or "semi-fix") - the solution that works for me is to enhance trpc init with transformer, like:

```js
export const t = initTRPC.create({
  transformer: superjson,
});
```

in theory it should not be needed to have types properly working, but in practice it solved my issue 🤔 .

**Reply by KATT** [reply]
Was likely that you're not using latest version of TS in your project or in your editor

https://trpc.io/docs/faq#it-doesnt-work-im-getting-any-everywhere
